pointcloud/basic_octree.go

```diff
@@ -182,3 +189,158 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// Pose returns the pose of the octree.
+func (octree *BasicOctree) Pose() spatialmath.Pose {
+	return octree.center
+}
+
+// AlmostEqual compares the octree with another geometry and checks if they are equivalent.
+func (octree *BasicOctree) AlmostEqual(geom spatialmath.Geometry) bool {
+	// TODO
+	return false
+}
+
+// Transform recursively steps through the octree and transforms it by the given pose.
+func (octree *BasicOctree) Transform(p spatialmath.Pose) spatialmath.Geometry {
+	var transformedOctree *BasicOctree
+	switch octree.node.nodeType {
+	case internalNode:
+		transformedOctree = &BasicOctree{
```

Comment:
It feels strange to declare `transformedOctree` and then do further manipulations on the object.
Why not do the math on `octree` and `octree.node`, etc.. and once there are no further manipulations to be made we declare and return the `transformedOctree`?

Reply:
won't that modify the octree that this function was called on? I was told that it's better to make a deep copy and do operations on that. transformedOctree is acting as the deep copy here so that we don't modify the original octree

Reply:
You're correct that we do not want to mutate the existing object.

I think what Nick is suggesting is that instead of initializing `transformedOctree.center` to `octree.center` and then mutating that, do the math ahead of time i.e. `newCenter := spatialmath.Compose(octree.center, p)` and then initialize `transformedOctree` with the correct new value. That way no octrees are being mutated, simply instantiated with correct values.

Reply:
got it, will do that